src/main/groovy/com/onesignal/androidsdk/GradleProjectPlugin.groovy

```diff
@@ -277,14 +276,39 @@ class GradleProjectPlugin implements Plugin<Project> {
             "${compileSdkVersion}.+"
         )
     }
-
+    
     static void overrideVersion(DependencyResolveDetails details, String resolvedVersion) {
+        // Omit NULLs
+        if (resolvedVersion == null)
+            return
+
+        // Omit override if never set a non-default version
         if (resolvedVersion == NO_REF_VERSION)
             return
 
+        // Omit override if no change
         if (details.requested.version == resolvedVersion)
             return
 
+        // Special Google rule if going from pre-15's non-semantic versions to 15+'s semantic versions
+        def newGoogleVersion = new ExactVersionSelector('15.0.0')
+        def isGoogleLibrary = details.requested.group == GROUP_GMS || details.requested.group == GROUP_FIREBASE
+        def hasAlignOver15 = isVersionInOrHigher(versionGroupAligns[GROUP_GMS]['version'] as String, newGoogleVersion) ||
+                             isVersionInOrHigher(versionGroupAligns[GROUP_FIREBASE]['version'] as String, newGoogleVersion)
+        if (isGoogleLibrary && hasAlignOver15) {
+            // These Google license libraries were dropped in 15.0.0+, omit them
+            if (details.requested.name ==~ /firebase-.*-license/ ||
+                details.requested.name ==~ /play-services-.*-license/)
+                return
+
+            // If the requested version is under 15 increase to version range that will include other libraries
+            if (isVersionBelow(details.requested.version, newGoogleVersion))
+                resolvedVersion = '[15.0.0, 16.0.0)'
```

Comment:
Brackets are mismatched here, is that correct?

Reply:
Brackets mean `including`, the parentheses mean `excluding`.

So essentially `[15.0.0, 16.0.0)` means "Any version including 15.0.0 up to but not including 16.0.0"